### PR TITLE
Feature/10464 crypto functions part 2/10641 requesting access token

### PR DESCRIPTION
**Pull request by nickguendling**
## Description
Implements the access token request

## Link to Jira
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-10641

### CODE REVIEW

File: src/xcode/ENA/ENA/Source/HTTPClient/Resources/__tests__/TicketValidationAccessTokenResourceTests.swift [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+
+		waitForExpectations(timeout: .short)
+	}
+
+	func testParseError() throws {
+		let expectation = expectation(description: "Expect that we got a completion")
+
+		let stack = MockNetworkStack(
+			httpStatus: 200,
+			responseData: "🦠".data(using: .ascii)
```
Comment (ArturFriesen) [inline src/xcode/ENA/ENA/Source/HTTPClient/Resources/__tests__/TicketValidationAccessTokenResourceTests.swift:45]: 😆

File: src/xcode/ENA/ENA/Source/Services/TicketValidation/AccessToken/TicketValidationAccessTokenProcessor.swift [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+	}
+
+	// MARK: - Internal
+	
+	func process(
+		jwtWithHeadersModel: JWTWithHeadersModel,
+		accessTokenSignJwkSet: [JSONWebKey],
+		completion: @escaping (Result<TicketValidationAccessTokenResult, TicketValidationAccessTokenProcessingError>) -> Void
+	) {
+		guard let nonceBase64 = jwtWithHeadersModel.headers["x-nonce"] as? String else {
```
Comment (ArturFriesen) [inline src/xcode/ENA/ENA/Source/Services/TicketValidation/AccessToken/TicketValidationAccessTokenProcessor.swift:24]: Could this be resolved at the network or resource level.
It seems a little odd that the processing needs to resolve the header.

File: src/xcode/ENA/ENA/Source/Services/TicketValidation/AccessToken/TicketValidationAccessTokenProcessor.swift [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+		switch jwtVerification.verify(jwtString: accessToken, against: accessTokenSignJwkSet) {
+		case .success:
+			guard let jwtObject = try? JWT<TicketValidationAccessToken>(jwtString: accessToken) else {
+				completion(.failure(.ATR_PARSE_ERR))
+				return
+			}
+
+			let accessTokenPayload = jwtObject.claims
+
+			guard accessTokenPayload.t == 1 || accessTokenPayload.t == 2 else {
```
Comment (ArturFriesen) [inline src/xcode/ENA/ENA/Source/Services/TicketValidation/AccessToken/TicketValidationAccessTokenProcessor.swift:41]: Would be nice to have the tech-spec entries as comments here for the different steps.

File: src/xcode/ENA/ENA/Source/Services/TicketValidation/AccessToken/TicketValidationAccessTokenProcessor.swift [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+	// MARK: - Init
+
+	init(jwtVerification: JWTVerifying) {
+		self.jwtVerification = jwtVerification
+	}
+
+	// MARK: - Internal
+	
+	func process(
```
Comment (ArturFriesen) [inline src/xcode/ENA/ENA/Source/Services/TicketValidation/AccessToken/TicketValidationAccessTokenProcessor.swift:19]: More logging would be nice. Maybe an info log when the processing starts and finished. And in between some debugging logs.

File: src/xcode/ENA/ENA/Source/Services/TicketValidation/Models/TicketValidationAccessToken.swift [hunk 5]
```diff
@@ -18,10 +19,10 @@ import Foundation
 // | `t` | int | The type of the validation (0 = Structure, 1 = Cryptographic, 2 = Full) |
 // | `vc` | object | A data structure representing the validation conditions (see below) |
 
-struct TicketValidationAccessToken: Codable {
+struct TicketValidationAccessToken: Equatable, Codable, Claims {
 	let iss: String
-	let iat: Int
-	let exp: Int
+	let iat: Date?
```
Comment (ArturFriesen) [inline src/xcode/ENA/ENA/Source/Services/TicketValidation/Models/TicketValidationAccessToken.swift:24]: Will this be decoded correctly from Int to Date?

File: src/xcode/ENA/ENA/Source/Services/TicketValidation/Models/TicketValidationServiceIdentityDocument.swift [hunk 6]
```diff
@@ -41,7 +41,7 @@ struct TicketValidationVerificationMethod: Codable {
 // | `serviceEndpoint` | string/url | URL to the service endpoint |
 // | `name` | string | Name of the service |
 
-struct TicketValidationValidationServiceData: Codable {
+struct TicketValidationServiceData: Codable {
```
Comment (ArturFriesen) [inline src/xcode/ENA/ENA/Source/Services/TicketValidation/Models/TicketValidationServiceIdentityDocument.swift:44]: 👍

**Review by ArturFriesen — CHANGES_REQUESTED** [review]

**Comment by KathrinLuecke** [comment]
There are no text changes in this PR so there is nothing for UA to review. What should I do now? Approve it anyhow?

File: src/xcode/ENA/ENA/Source/Services/TicketValidation/Models/TicketValidationAccessToken.swift (hunk 5 above)
Comment (KaiTeuber) [reply src/xcode/ENA/ENA/Source/Services/TicketValidation/Models/TicketValidationAccessToken.swift:24]: We checked the decoder, dateDecodingStrategy is set correct.

**Review by CV113 — APPROVED** [review]
UA approved

**Review by KaiTeuber — APPROVED** [review]

**Review by ArturFriesen — APPROVED** [review]

**Pull request merged**